web/src/store/index.js

```diff
@@ -166,6 +174,27 @@ const mutations = {
       },
     });
   },
+  /**
+   * @private
+   */
+  [SET_VALIDATED_DATASET_DATA](state, { data }) {
+    const validatedMeasurements = parsePandasDataFrame(data.measurements);
+    const validatedGroups = parsePandasDataFrame(data.groups);
+    const validatedMeasurementsMetaData = parsePandasDataFrame(data.measurement_metadata);
+    const validatedSampleMetaData = parsePandasDataFrame(data.sample_metadata);
+    const ds = state.datasets[data.csv_file_id];
+
+    const delta = {
+      validatedMeasurements,
+      validatedGroups,
+      validatedMeasurementsMetaData,
+      validatedSampleMetaData,
+    };
```

Comment:
Prefer not to introduce new keys into the dataset object like this.  You'll end up with cases where the dataset exists but these keys don't.  It's been my goal to guarantee the existence of all keys as soon as the dataset is initiated.

Easiest fix would be to add these new properties into `datasetDefaults` with `null` or `[]`  as defaults or whatever you want.